File: 007/2023_11_25/abc115c.jl
N, K = broadcast(n -> parse(Int, n), split(readline()))
hN = []

for i in 1:N
    push!(hN, parse(Int, readline()))
end

answer = 100000000
hN = sort(hN)

for i in 1:N-K+1
    current_min = hN[i+K-1] - hN[i]

    if current_min ≤ answer
        global answer = current_min
    end
end

println(answer)

File: 01/003/2023_10_30/abc115c.py
N, K = map(int, input().split())
Hn = []

for i in range(N):
    Hn.append(int(input()))

Hn.sort()

answer = 10**9

for i in range(0, N - K + 1):
    diff = Hn[i + K - 1] - Hn[i]

    if answer >= diff:
        answer = diff

print(answer)
